### PR TITLE
add spine customMaterial & set default UI depthTest False

### PR DESCRIPTION
**Pull request by LinYunMo**
Re: https://github.com/cocos-creator/3d-tasks/issues/6139

### CODE REVIEW

**Review by PatriceJiang — APPROVED** [review]

**Review by SantyWang — APPROVED** [review]

File: cocos/spine/skeleton.ts [hunk 1]
```diff
@@ -132,12 +131,15 @@ export class Skeleton extends Renderable2D {
     get meshRenderDataArray () { return this._meshRenderDataArray; }
 
     @override
-    @visible(false)
+    @type(Material)
+    @displayOrder(0)
+    @displayName('CustomMaterial')
     get customMaterial () {
-        return super.customMaterial;
```
Comment (caryliu1999) [inline cocos/spine/skeleton.ts:137]: 组件不是有通用的材质属性吗，为什么这里要单独再设置一个属性来指定？

**Review by caryliu1999 — APPROVED** [review]

File: cocos/spine/skeleton.ts [hunk 2]
```diff
@@ -1638,6 +1638,16 @@ export class Skeleton extends Renderable2D {
             }
         });
     }
+
+    private _cleanMaterialCache () {
+        const cache = Object.values(this._materialCache);
```
Comment (pandamicro) [inline cocos/spine/skeleton.ts]: Should use `for in` here, don't create array

File: editor/assets/effects/builtin-spine.effect [hunk 3]
```diff
@@ -22,71 +22,60 @@ CCEffect %{
...
+    in vec4 a_color2;
+    out vec4 v_dark;
+  #endif
+
   vec4 vert () {
     vec4 pos = vec4(a_position, 1);
 
-    #if USE_LOCAL
-      pos = cc_matWorld * pos;
-    #endif
```
Comment (pandamicro) [inline editor/assets/effects/builtin-spine.effect:43]: Should keep USE_LOCAL define, we might use model matrix for spine, if there are no many spine copies, using model matrix would be much better than transform all vertices to world coordinates

File: editor/assets/effects/builtin-spine.effect [hunk 4]
```diff
@@ -22,71 +22,60 @@ CCEffect %{
...
-        o.r = o.g = o.b = gray;
-      #endif
-    #endif
+    vec4 o = vec4(1, 1, 1, 1);
+    #if TWO_COLORED
+      vec4 texColor = vec4(1, 1, 1, 1);
 
- 	  o.a = texColor.a * v_light.a;
-    o.rgb = ((texColor.a - 1.0) * v_dark.a + 1.0 - texColor.rgb) * v_dark.rgb + texColor.rgb * v_light.rgb;
+      texColor *= texture(cc_spriteTexture, uv0);
```
Comment (pandamicro) [inline editor/assets/effects/builtin-spine.effect:79]: Should use CCSampleTexture

**Review by pandamicro — CHANGES_REQUESTED** [review]

**Review by pandamicro — APPROVED** [review]

**Pull request merged**